### PR TITLE
added revoke token test and moved auth manager tests in own file

### PR DESCRIPTION
**Pull request by njarecha**

### CODE REVIEW

File: Tests/ArcGISToolkitTests/AuthenticationManagerTests.swift [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+            )
+        )
+        
+        var arcGISCreds = ArcGISEnvironment.authenticationManager.arcGISCredentialStore.credentials
+        XCTAssertEqual(arcGISCreds.count, 1)
+        
+        await ArcGISEnvironment.authenticationManager.clearCredentialStores()
+        
+        arcGISCreds = ArcGISEnvironment.authenticationManager.arcGISCredentialStore.credentials
+        XCTAssertTrue(arcGISCreds.isEmpty)
```
Comment (dfeinzimer) [inline Tests/ArcGISToolkitTests/AuthenticationManagerTests.swift:63]: Would it be better to test `ArcGISEnvironment.authenticationManager.arcGISCredentialStore.credentials` directly?

```suggestion
        XCTAssertEqual(ArcGISEnvironment.authenticationManager.arcGISCredentialStore.credentials.count, 1)
        
        await ArcGISEnvironment.authenticationManager.clearCredentialStores()
        
        XCTAssertTrue(ArcGISEnvironment.authenticationManager.arcGISCredentialStore.credentials.isEmpty)
```

**Review by dfeinzimer — DISMISSED** [review]

File: Tests/ArcGISToolkitTests/AuthenticationManagerTests.swift [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    func testHandleChallengesUsingAuthenticator() {
+        addTeardownBlock {
+            ArcGISEnvironment.authenticationManager.arcGISAuthenticationChallengeHandler = nil
+            ArcGISEnvironment.authenticationManager.networkAuthenticationChallengeHandler = nil
+        }
+        
+        let authenticator = Authenticator()
+        ArcGISEnvironment.authenticationManager.handleChallenges(using: authenticator)
+        XCTAssertNotNil(ArcGISEnvironment.authenticationManager.arcGISAuthenticationChallengeHandler)
+        XCTAssertNotNil(ArcGISEnvironment.authenticationManager.networkAuthenticationChallengeHandler)
```
Comment (rolson) [inline Tests/ArcGISToolkitTests/AuthenticationManagerTests.swift:28]: you can assert that it's equal to the authenticator, right?

Comment (njarecha) [reply Tests/ArcGISToolkitTests/AuthenticationManagerTests.swift:28]: The handles are not `Equatable`. Shall we make them?

Comment (rolson) [reply Tests/ArcGISToolkitTests/AuthenticationManagerTests.swift:28]: no but you can cast to an authenticator I think with `as?` right?

Comment (njarecha) [reply Tests/ArcGISToolkitTests/AuthenticationManagerTests.swift:28]: yes, I can do `XCTAssertIdentical` using `as?`.

```
        XCTAssertIdentical(ArcGISEnvironment.authenticationManager.arcGISAuthenticationChallengeHandler as? Authenticator, authenticator)
        XCTAssertIdentical(ArcGISEnvironment.authenticationManager.networkAuthenticationChallengeHandler as? Authenticator, authenticator)
```

**Review by rolson — APPROVED** [review]

**Review by dfeinzimer — APPROVED** [review]

**Comment by njarecha** [comment]
Thanks @rolson and @dfeinzimer!

**Pull request merged**